security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -559,6 +562,9 @@ func (sc *SecretManagerClient) generateNewSecret(resourceName string) (*security
 	numOutgoingRequests.With(RequestType.Value(monitoring.CSR)).Increment()
 	timeBeforeCSR := time.Now()
 	certChainPEM, err := sc.caClient.CSRSign(csrPEM, int64(sc.configOptions.SecretTTL.Seconds()))
+	if err == nil {
+		trustBundlePEM, err = sc.caClient.GetRootCertBundle()
```

Comment:
I think this technically has a race condition - if root root cert bundle changes since CSRSign is done. But probably not a practical concern..

Reply:
IMO, its okay to have an implicit assumption that a new CA's trustAnchor will always be returned by the trustAnchor API call before its allowed to sign a cert.